daemon/logger/syslog/syslog.go

```diff
@@ -0,0 +1,61 @@
+package syslog
+
+import (
+	"log/syslog"
+	"os"
+	"path"
+	"sync"
+
+	"github.com/docker/docker/daemon/logger"
+)
+
+type Syslog struct {
+	writer *syslog.Writer
+	tag    string
+	mu     sync.Mutex
+}
+
+func New(tag string) (logger.Logger, error) {
+	return &Syslog{
+		writer: nil,
+		tag:    tag,
+	}, nil
+}
+
+func (s *Syslog) Log(msg *logger.Message) error {
+	s.mu.Lock()
```

Comment:
Why do you need a lock here?  I don't think hitting a lock every time you want to write a log line is good. 

You should initialize the logger in the New() func, now lazily on write.  If you do this I don't think the lock is required.